core_impl/src/main/java/io/opencensus/trace/SpanImpl.java

```diff
@@ -34,7 +36,7 @@
 
 /** Implementation for the {@link Span} class. */
 @ThreadSafe
-final class SpanImpl extends Span {
+public final class SpanImpl extends Span {
```

Comment:
instead of doing this, you could make a Internal package accessor type. like `InternalExport.startSpan(...)` which returns span. This makes it blatantly obvious what's for internal use and not.